backend/resource/backend.go

```diff
@@ -0,0 +1,128 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/messaging"
+	"github.com/sensu/sensu-go/backend/store"
+	"github.com/sensu/sensu-go/system"
+	"github.com/sensu/sensu-go/types"
+)
+
+type eventInfo struct {
+	status       uint32
+	timestampSec int64
+}
+
+type BackendResource struct {
+	nsStore           store.NamespaceStore
+	entityStore       store.EntityStore
+	bus               messaging.MessageBus
+	backendEntity     *corev2.Entity
+	lastEvents        map[string]*eventInfo
+	repeatIntervalSec int64
+}
+
+const (
+	// ComponentSecrets represents the secrets component of the Sensu backend
+	ComponentSecrets = "secrets"
+
+	// The default Sensu system namespace
+	systemNamespaceName = "sensu-system"
+)
+
+func New(nsStore store.NamespaceStore, entityStore store.EntityStore, bus messaging.MessageBus) *BackendResource {
+	return &BackendResource{
+		nsStore:           nsStore,
+		entityStore:       entityStore,
+		bus:               bus,
+		lastEvents:        map[string]*eventInfo{},
+		repeatIntervalSec: int64(30),
+	}
+}
+
+func (br *BackendResource) EnsureBackendResources(ctx context.Context) error {
+	ns, err := br.nsStore.GetNamespace(ctx, systemNamespaceName)
+	if err != nil {
+		return err
+	}
+
+	if ns == nil {
+		err = br.nsStore.CreateNamespace(ctx, &types.Namespace{Name: systemNamespaceName})
+		if err != nil {
+			return err
+		}
+	}
+
+	backendEntity, err := getEntity()
+	if err != nil {
+		return err
+	}
+	err = br.entityStore.UpdateEntity(ctx, backendEntity)
+	if err != nil {
+		return err
+	}
+
+	br.backendEntity = backendEntity
+
+	return nil
+}
+
+func (br *BackendResource) GenerateBackendEvent(component string, status uint32, output string) error {
+	if br.backendEntity == nil {
+		return errors.New("backend entity doesn't exist")
+	}
+
+	now := time.Now().Unix()
+	checkName := fmt.Sprintf("%s-%s", corev2.BackendCheckName, component)
```

Comment:
I don't think prepending all the backend event check names with "backend" is necessary. We already have "backend" as the entity name, which is part of the key to retrieve an event.

Reply:
Agreed, fixed.